wgpu-hal/src/vulkan/device.rs

```diff
@@ -742,6 +742,15 @@ impl crate::Device<super::Api> for super::Device {
         &self,
         desc: &crate::BufferDescriptor,
     ) -> Result<super::Buffer, crate::DeviceError> {
+        if self
+            .shared
+            .workarounds
+            .contains(super::Workarounds::BUFFER_EXTENTS_I32)
```

Comment:
Do we already advertise this in the max buffer binding size?

Reply:
Hm, that commit wasn't supposed to show up in ths PR.